File: sdk/python/hitechcloud/resources/hosting.py
"""Hosting Services resource (6 endpoints)"""

from .base import BaseResource

class HostingResource(BaseResource):
    """Hosting Services: reinstall, status, bandwidth, reset"""

    def get_reinstall_details(self, service_id: int) -> dict:
        """GET /api/service/{id}/reinstall - Return installation options"""
        return self._http.get(f"/api/service/{service_id}/reinstall")

    def reinstall(self, service_id: int, **kwargs) -> dict:
        """POST /api/service/{id}/reinstall - Reinstall server"""
        return self._http.post(f"/api/service/{service_id}/reinstall", kwargs)

    def get_status(self, service_id: int) -> dict:
        """GET /api/service/{id}/status - Return server power status"""
        return self._http.get(f"/api/service/{service_id}/status")

    def get_bandwidth_graphs(self, service_id: int) -> dict:
        """GET /api/service/{id}/bandwidth-graphs - Return bandwidth usage graphs"""
        return self._http.get(f"/api/service/{service_id}/bandwidth-graphs")

    def get_bandwidth_usage(self, service_id: int) -> dict:
        """GET /api/service/{id}/bandwidth - Return bandwidth usage and billing"""
        return self._http.get(f"/api/service/{service_id}/bandwidth")

    def reset(self, service_id: int) -> dict:
        """POST /api/service/{id}/reset - Reset server"""
        return self._http.post(f"/api/service/{service_id}/reset")

File: sdk/ruby/lib/hitechcloud/resources/hosting.rb
module HiTechCloud
  # Hosting resource
  class HostingResource < BaseResource

    # Get reinstall info
    def reinstall(id)
      @http.get("/api/service/%{id}/reinstall" % { id: id })
    end

    # Reinstall
    def do_reinstall(id, data)
      @http.post("/api/service/%{id}/reinstall" % { id: id }, data)
    end

    # Get status
    def status(id)
      @http.get("/api/service/%{id}/status" % { id: id })
    end

    # Get bandwidth graphs
    def bandwidth_graphs(id)
      @http.get("/api/service/%{id}/bandwidth-graphs" % { id: id })
    end

    # Get bandwidth
    def bandwidth(id)
      @http.get("/api/service/%{id}/bandwidth" % { id: id })
    end

    # Reset password
    def reset_password(id, data)
      @http.post("/api/service/%{id}/reset" % { id: id }, data)
    end

  end
end
